src/_pytest/compat.py

```diff
@@ -357,36 +290,6 @@ def safe_isclass(obj):
         return False
 
 
-def _is_unittest_unexpected_success_a_failure():
-    """Return if the test suite should fail if an @expectedFailure unittest test PASSES.
-
-    From https://docs.python.org/3/library/unittest.html?highlight=unittest#unittest.TestResult.wasSuccessful:
-        Changed in version 3.4: Returns False if there were any
-        unexpectedSuccesses from tests marked with the expectedFailure() decorator.
-    """
-    return sys.version_info >= (3, 4)
-
-
-if _PY3:
-
-    def safe_str(v):
```

Comment:
i noted this one is evil, looking at usage sites made me think the removal was bad in some case

Reply:
What do you mean @RonnyPfannschmidt? The Python 3 version was just calling `str()`, so I changed the call sites to call `str()` as well.

Reply:
@nicoddemus the function name was terrible for indicating the actual intent

Reply:
Oh OK.

Could you please "accept" the PR once you finished the review? It would be nice to get this into `master` soon.